keyboards/nullbitsco/snap/keymaps/iso/keymap.c

```diff
@@ -0,0 +1,104 @@
+    KC_F13,  KC_GRV,   KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,     KC_7,   KC_8,   KC_9,    KC_0,      KC_MINS,  KC_EQL,        KC_BSPC,       KC_HOME,
+    KC_F14,  KC_TAB,   KC_Q,    KC_W,   KC_E,   KC_R,           KC_T,     KC_Y,   KC_U,   KC_I,    KC_O,      KC_P,     KC_LBRC,  KC_RBRC,  KC_ENT,   KC_END,
+    KC_F15,  KC_CAPS,  KC_A,    KC_S,   KC_D,   KC_F,           KC_G,     KC_H,   KC_J,   KC_K,    KC_L,      KC_SCLN,  KC_QUOT,  KC_BSLS,            KC_PGUP,
+    KC_F16,  KC_LSFT,  KC_NUHS, KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,     KC_N,   KC_M,   KC_COMM, KC_DOT,    KC_SLSH,  KC_RSFT,            KC_UP,    KC_PGDN,
+    KC_F17,  KC_LCTL,  KC_LGUI, KC_LALT,        MO(_FUNC),     KC_SPC,   KC_SPC,                   MO(_FUNC), KC_RALT,  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
+  ),
+    [_FUNC] = LAYOUT_iso(
+             RESET,   _____,  _____,  _____,  _____,  _____,    _____,  _____,  _____,   _____,     _____,    _____,    _____,    _____,    _____,
+    RGB_TOG, _____,   _____,  _____,  _____,  _____,  _____,    _____,  _____,  _____,   _____,     _____,    _____,    _____,         _____,        _____,
+    _____,   _____,   _____,   _____,  _____,  _____,          _____,    _____,  _____,  _____,   _____,     _____,    _____,    _____,    _____,    _____,
+    _____,   _____,   _____,   _____,  _____,  _____,          _____,    _____,  _____,  _____,   _____,     _____,    _____,    _____,              _____,
+    _____,   _____,   _____,   _____,  _____,  _____,  _____,  _____,    _____,  _____,  _____,   _____,     _____,    _____,              _____,    _____,
+    _____,   _____,   _____,   _____,          _____,          _____,    _____,                   _____,     _____,    _____,    KC_MPRV,  KC_MPLY,  KC_MNXT
+  ),
+};
+// clang-format on
+
+// RGB config, for changing RGB settings on non-VIA firmwares
+#ifdef RGBLIGHT_ENABLE
+void change_RGB(bool clockwise) {
+    bool shift = get_mods() & MOD_MASK_SHIFT;
+    bool alt   = get_mods() & MOD_MASK_ALT;
+    bool ctrl  = get_mods() & MOD_MASK_CTRL;
+
+#ifdef CONSOLE_ENABLE
+    dprintf("Mods: %u\n", get_mods());
+#endif
+
+    if (clockwise) {
+        if (alt) {
+            rgblight_increase_hue();
+        } else if (ctrl) {
+            rgblight_increase_val();
+        } else if (shift) {
+            rgblight_increase_sat();
+        } else {
+            rgblight_step();
+        }
+
+    } else {
+        if (alt) {
+            rgblight_decrease_hue();
+        } else if (ctrl) {
+            rgblight_decrease_val();
+        } else if (shift) {
+            rgblight_decrease_sat();
+        } else {
+            rgblight_step_reverse();
+        }
+    }
+}
+#endif
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (layer_state_is(1)) {
+        // change RGB settings
+        #ifdef RGBLIGHT_ENABLE
+        change_RGB(clockwise);
```

Comment:
just a heads up, encoder map supports rgb keycodes, without the need for a separate function/handling.